### PR TITLE
feat(scm): add bitbucket cloud scm and action support

### PR DESCRIPTION
**Pull request by mcwarman**
Add support for Bitbucket Cloud:

* a new scm `bitbucket`
* a new action `bitbucket/pullrequest`

Also updates documentation for stash to make it clearer its Bitbucket Server.

## Test

To test this pull request, you can run the following commands:

```shell
go build -o bin/updatecli
./bin/updatecli diff --config ./e2e/updatecli.d/success.d/bitbucket/scm.yaml
```

### Tradeoff

Bitbucket doesn't support any HTML in PR descriptions, so I had to add Markdown only views. There's a potential the format could be improved going forward

### Potential improvement

Markdown format of PRs.

### CODE REVIEW

**Comment by mcwarman** [comment]
@olblak if you want to set-up a repo in bitbucket that you control, I'm happy to update the e2e tests to use that instead.

**Comment by olblak** [comment]
Ho nice 

> Bitbucket doesn't support any HTML in PR descriptions, so I had to add Markdown only views. There's a potential the format could be improved going forward

It's unfortunate because using HTML allows Updatecli to store information in <div> so it can detect when a pullrequest body needs to be updated. 

> if you want to set-up a repo in bitbucket that you control, I'm happy to update the e2e tests to use that instead.

I think it would make sense to create a shared bitbucket instance as we can have up to 5 seats
https://www.atlassian.com/software/bitbucket/pricing

**Comment by mcwarman** [comment]
> It's unfortunate because using HTML allows Updatecli to store information in
> so it can detect when a pullrequest body needs to be updated.

Yeah I've started using Bitbucket in my new role, its not my favourite user experience.

It also doesn't support HTML comments so we couldn't get clever with hiding information in them.

**Comment by olblak** [comment]
@mcwarman I just open a pullrequest on your fork, to update the e2e tests
https://github.com/mcwarman/updatecli/pull/1

Apparently, I already created a Bitbucket account, and we can have up to 5 people on that workspace

**Comment by olblak** [comment]
Could I ask to also add a documentation page to the website next to https://github.com/updatecli/website/tree/master/content/en/docs/plugins/scm

**Comment by mcwarman** [comment]
I don't have access to see that repo on bitbucket (if you added `mcwarman` that's an old work account I can't get back the username for `mcwarman1` is now me).

I'll add the docs for using it.

**Comment by mcwarman** [comment]
We either need to make the test repo public, or set BITBUCKET_TOKEN for the e2e generated from the repo:

Ref: https://support.atlassian.com/bitbucket-cloud/docs/access-tokens/

**Comment by olblak** [comment]
> We either need to make the test repo public, or set BITBUCKET_TOKEN for the e2e generated from the repo:
> 
> Ref: https://support.atlassian.com/bitbucket-cloud/docs/access-tokens/

Well I thought it already made it public, I'll check that later

File: pkg/plugins/resources/bitbucket/pullrequest/main.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+	scm *bitbucket.Bitbucket
+	// SourceBranch specifies the pullrequest source branch.
+	SourceBranch string `yaml:",inline,omitempty"`
+	// TargetBranch specifies the pullrequest target branch
+	TargetBranch string `yaml:",inline,omitempty"`
+	// Owner specifies repository owner
+	Owner string `yaml:",omitempty" jsonschema:"required"`
+	// Repository specifies the name of a repository for a specific owner
+	Repository string `yaml:",omitempty" jsonschema:"required"`
+}
```
Comment (mcwarman) [inline pkg/plugins/resources/bitbucket/pullrequest/main.go:43]: Something isn't quite correct here, going to have a look.

**Review by olblak — APPROVED** [review]
Thanks for the pull request, it looks great.

**Pull request merged**